sudo/src/run_handler.rs

```diff
@@ -191,19 +191,58 @@ pub fn run_target(
     do_request(req, copy_env, manually_requested_dir)
 }
 
-fn search_path(filename: &str) -> Result<String> {
-    let filename = &HSTRING::from(filename);
-    let len = unsafe { SearchPathW(None, filename, None, None, None) };
+/// Returns all the extensions from PATHEXT that are used for "executables"
+fn get_path_extensions() -> Vec<String> {
+    env::var("PATHEXT")
+        .unwrap_or_else(|_| {
+            // If PATHEXT isn't set, use the default
+            ".COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC".to_string()
+        })
+        .split(';')
+        .map(|ext| ext.to_string())
+        .collect()
+}
```

Comment:
If this was inside `search_path_with_extensions` it wouldn't need to collect into a vector first, nor allocate the default value (although that shouldn't happen since PATHEXT should always exist).

Like this:
```rust
let pathext = env::var("PATHEXT");
let extensions = pathext
    .as_deref()
    // If PATHEXT isn't set, use the default
    .unwrap_or(".COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC")
    .split(';');
```

Reply:
Thinking about it some more, I think it'd be preferable if we still did this, but it's not something I would block over.